src/AWS.Deploy.Common/Recipes/Validation/RecipeValidators/DockerfilePathValidator.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.IO;
+using AWS.Deploy.Common.IO;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// This validates that the Dockerfile is within the build context.
+    ///
+    /// Per https://docs.docker.com/engine/reference/commandline/build/#text-files
+    /// "The path must be to a file within the build context."
+    /// </summary>
+    public class DockerfilePathValidator : IRecipeValidator
+    {
+        public ValidationResult Validate(Recommendation recommendation, IDeployToolValidationContext deployValidationContext)
+        {
+            IDirectoryManager directoryManager = new DirectoryManager();
```

Comment:
IDirectoryManager should be passed using dependency injection and not instantiated here.

Reply:
I think similar to `IOptionSettingItemValidator`, we're not set up to inject into recipe validators because they're designed to be constructed without any parameters.

https://github.com/aws/aws-dotnet-deploy/blob/f2e5db538a032be70de1dc3953b297528a62afeb/docs/OptionSettingsItems-input-validation.md?plain=1#L142-L146

Reply:
Switched to injection now that it's possible after https://github.com/aws/aws-dotnet-deploy/pull/560 has been merged.